kolibri/core/content/views.py

```diff
@@ -319,3 +322,30 @@ def get(self, request, filename, new_filename):
         response["Content-Length"] = os.path.getsize(path)
 
         return response
+
+
+class ContentPermalinkRedirect(View):
+
+    def get(self, request, *args, **kwargs):
+
+        # extract the GET parameters
+        channel_id = request.GET.get("channel_id")
+        node_id = request.GET.get("node_id")
+        content_id = request.GET.get("content_id")
+
+        try:  # first, try to get the node by the unique node_id
+            node = ContentNode.objects.get(id=node_id)
+        except ContentNode.DoesNotExist:  # if it isn't found, fall back to looking for the content_id in the channel
+            node = ContentNode.objects.filter(channel_id=channel_id, content_id=content_id).first()
+            if not node:  # if it's still not found, see if we can find anything with the content_id across any channel
+                node = ContentNode.objects.filter(content_id=content_id).first()
+
+        # build up the target topic/content page URL
+        if node:
+            url = None
+            for hook in ContentNodeDisplayHook().registered_hooks:
+                url = hook.node_url(node)
+            if url:
+                return HttpResponseRedirect(url)
```

Comment:
HttpResponseRedirect = 302 = "Moved Temporarily"
using HttpResponsePermanentRedirect would result in 301 

I have never seen a HTTP system where the difference mattered so probably either is good, but wanted to flag the two options for your considerations.

Reply:
Agreed -- one place it can matter is caching, I think.